Document auto-migrate feature (#129)
rancher/turtles-docs: docs/tasks/maintenance/automigrate_to_v3_import.md

---
sidebar_position: 3
---

# Auto-migration to v3 cluster import

:::caution

> Please read the step described in [import controller upgrade](./import_controller_upgrade.md#context), and consider how you want to perform your migration. Feature described here will automatically migrate all old cluster in one go, replacing them with v3 cluster objects.

Reviewer: ```suggestion
Please read the step described in [import controller upgrade](./import_controller_upgrade.md#context), and consider how you want to perform your migration. Feature described here will automatically migrate all old clusters in one go, replacing them with v3 cluster objects.
```

:::

## Context

> In order to facilitate migration between provisioning v1 and management v3 cluster, Turtles provide a separate feature flag named `managementv3-cluster-migration`. This option is disabled by default, but can be enabled during installation by enabling both checkmark in the list:

Reviewer: ```suggestion
In order to facilitate migration between provisioning v1 and management v3 cluster, Turtles provides a separate feature flag named `managementv3-cluster-migration`. This option is disabled by default, but can be enabled during installation by enabling both checkmarks in the list:
```

![automigrate feature](image.png)

By default the feature is disabled.

## Functionality

> Intended behavior for the feature flag is described in the [ADR](https://github.com/rancher/turtles/blob/main/docs/adr/0011-v1-to-v3-migration.md). In short, having the flag set will prevent duplication of cluster imports on upgrade to `management/v3` based import or downgrade (upgrade) to opt-out of `management/v3` cluster, back to `provisioning/v1`.

Reviewer 2: Reading this:

```
In short, having the flag set will prevent duplication of cluster imports on upgrade to `management/v3` based import or downgrade (upgrade) to opt-out of `management/v3` cluster, back to `provisioning/v1`.
```

confuses me a bit tbh, can we somehow make it simpler to understand..

> Intended behavior for the feature flag is described in the [ADR](https://github.com/rancher/turtles/blob/main/docs/adr/0011-v1-to-v3-migration.md). In short, having the flag set will prevent duplication of cluster imports on upgrade to `management/v3` based import or downgrade (upgrade) to opt-out of `management/v3` cluster, back to `provisioning/v1`.

Reviewer 2: ```suggestion
Intended behavior for the feature flag is described in the [ADR](https://github.com/rancher/turtles/blob/main/docs/adr/0011-v1-to-v3-migration.md). In short, having the flag set will prevent duplication of cluster imports on upgrade to `management/v3` based import or downgrade (upgrade) to opt-out of `management/v3` cluster, back to `provisioning/v1`.
```
